rs/nns/cmc/src/main.rs

```diff
@@ -3158,4 +3208,138 @@ mod tests {
         )
         .expect("The CMC canister interface is not compatible with the cmc.did file");
     }
+
+    #[test]
+    fn test_transaction_has_expected_memo_happy() {
+        // Not relevant to this test.
+        let operation = Operation::Mint {
+            to: AccountIdentifier::new(PrincipalId::new_user_test_id(668_857_347), None),
+            amount: Tokens::from_e8s(123_456),
+        };
+
+        // Case A: Legacy memo is used.
+        let transaction_with_legacy_memo = Transaction {
+            memo: Memo(42),
+            icrc1_memo: None,
+
+            // Irrelevant to this test.
+            operation: operation.clone(),
+            created_at_time: None,
+        };
+
+        assert_eq!(
+            transaction_has_expected_memo(&transaction_with_legacy_memo, Memo(42),),
+            Ok(()),
+        );
+
+        // Case B: icrc1_memo is used.
+        let transaction_with_icrc1_memo = Transaction {
+            memo: Memo(0),
+            icrc1_memo: Some(ByteBuf::from(vec![43, 0, 0, 0, 0, 0, 0, 0])),
```

Comment:
Is there no standard way to build icrc1 memos? I mean, this method seems pretty fragile since a human needs to count the number of trailing zeros, of which there must be exactly 7.